extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -1196,4 +1334,47 @@ public static String unescapeDocument(@Nonnull final String doc) {
                 .replaceAll("\\\\x5b", "[")
                 .replaceAll("\\\\x5d", "]");
     }
+
+    /**
+     * Generate a content playback nonce (also called {@code cpn}), sent by YouTube clients in
+     * playback requests (and also for some clients, in the player request body).
+     *
+     * @return a content playback nonce string
+     */
+    @Nonnull
+    public static String generateContentPlaybackNonce() {
+        final SecureRandom random = new SecureRandom();
+        final StringBuilder stringBuilder = new StringBuilder();
+
+        for (int i = 0; i < 16; i++) {
+            stringBuilder.append(CONTENT_PLAYBACK_NONCE_ALPHABET.charAt(
+                    (random.nextInt(128) + 1) & 63));
```

Comment:
What? Shouldn't you just select a random character from the alphabet? Why generate a number with at most 7 bits (128 = 2^7 = 10000000 bin), then you sum one, then you keep only the last 6 bits by doing & 63 (63 = 111111 bin). The 'A' will never be obtained because of the `+1`. What's wrong with just using `random.nextInt(CONTENT_PLAYBACK_NONCE_ALPHABET.size())`? Also, I'd create a method in `Utils` called `generateRandomStringFromAlphabet(final String alphabet, final int length)`, so that code is not duplicated.